src/main/java/no/nav/familie/inntektsmelding/forespørsel/modell/ForespørselOgSakRepository.java

```diff
@@ -0,0 +1,152 @@
+import jakarta.inject.Inject;
+import jakarta.persistence.EntityManager;
+import no.nav.familie.inntektsmelding.koder.ForespørselStatus;
+import no.nav.familie.inntektsmelding.koder.SakStatus;
+import no.nav.familie.inntektsmelding.koder.Ytelsetype;
+import no.nav.familie.inntektsmelding.typer.entitet.AktørIdEntitet;
+
+@Dependent
+public class ForespørselOgSakRepository {
+
+    private EntityManager entityManager;
+
+    public ForespørselOgSakRepository() {
+    }
+
+    @Inject
+    public ForespørselOgSakRepository(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+
+    public SakEntitet opprettSak(Ytelsetype ytelsetype, String aktørId, String orgnummer, String fagsakSaksnummer) {
+        var sakEntitet = new SakEntitet(orgnummer, new AktørIdEntitet(aktørId), ytelsetype, fagsakSaksnummer);
+        entityManager.persist(sakEntitet);
+        entityManager.flush();
+        return sakEntitet;
+    }
+
+    public UUID lagreForespørsel(LocalDate skjæringstidspunkt, Ytelsetype ytelsetype, String aktørId, String orgnummer, String fagsakSaksnummer) {
+        var sak = finnSakUnderBehandling(new AktørIdEntitet(aktørId), ytelsetype, orgnummer, fagsakSaksnummer);
+        if (sak.isEmpty()) {
+            throw new IllegalStateException("Kan ikke opprette forespørsel uten å ha en åpen sak");
+        }
+
+        var forespørselEntitet = new ForespørselEntitet(sak.get(), skjæringstidspunkt);
+
+        entityManager.persist(forespørselEntitet);
+        entityManager.flush();
+        return forespørselEntitet.getUuid();
+    }
+
+    public void oppdaterOppgaveId(UUID forespørselUUID, String oppgaveId) {
+        var forespørselOpt = hentForespørsel(forespørselUUID);
+        if (forespørselOpt.isPresent()) {
+            var forespørsel = forespørselOpt.get();
+            forespørsel.setOppgaveId(oppgaveId);
+            entityManager.persist(forespørsel);
+            entityManager.flush();
+        }
+    }
+
+    public void oppdaterSakId(Long internSakId, String sakId) {
+        var sak = hentSak(internSakId);
+        sak.setFagerSakId(sakId);
+        entityManager.persist(sak);
+        entityManager.flush();
+    }
+
+    public Optional<ForespørselEntitet> hentForespørsel(UUID uuid) {
+        var query = entityManager.createQuery("FROM ForespørselEntitet where uuid = :foresporselUUID", ForespørselEntitet.class)
+            .setParameter("foresporselUUID", uuid);
+
+        var resultList = query.getResultList();
+        if (resultList.isEmpty()) {
+            return Optional.empty();
+        } else if (resultList.size() > 1) {
+            throw new IllegalStateException("Forventet å finne kun en forespørsel for oppgitt uuid " + uuid);
+        } else {
+            return Optional.of(resultList.getFirst());
+        }
+    }
+
+    public void ferdigstillSak(Long internSakId) {
+        var resultat = hentSak(internSakId);
+        resultat.setSakStatus(SakStatus.FERDIG);
```

Comment:
Må vi ha en egen status for Sak egentlig? Vi kunne da kanskje sjekke om alle Forespørsel er løst?

Reply:
Tanken min no er at fagsystemet bestemmer når saken skal ferdigstilles, men vi kan gjerne ta ein diskusjon på det.

Reply:
Det blir fort styr med lukking, gjenåpning etc. Litt usikker om det gir så mye verdi - Saken forsvinner ikke fra Portalen etter den er lukket heler

Reply:
Vi skal vel ikkje gjenåpne saker? Eg har tatt utgangspunkt i at ein sak følger ein behandling i k9 og at vi oppretter ein ny sak dersom det kjem inn nye søknader.